### PR TITLE
DOC, MAINT: HTTP -> HTTPS, and other linkrot fixes

### PR DESCRIPTION
**Pull request by mwtoews**
Most of the fixes here upgrade from HTTP to HTTPS, particularly for Numpy and Scipy pages.

Other modifications are to update websites, as they exist today. Some older websites are revived using https://web.archive.org and few others that have no archive were marked with "(dead link)". These should be considered for either removal or replacement.

Some minor modifications were also done to NEPs; I can undo these if they need to be preserved.

### CODE REVIEW

**Review by eric-wieser — APPROVED** [review]

File: doc/HOWTO_RELEASE.rst.txt [hunk 1]
```diff
@@ -18,16 +18,16 @@ Source tree
 NumPy Docs
 ----------
 * https://github.com/numpy/numpy/blob/master/doc/HOWTO_RELEASE.rst.txt
-* http://projects.scipy.org/numpy/wiki/MicrosoftToolchainSupport
+* http://projects.scipy.org/numpy/wiki/MicrosoftToolchainSupport (dead link)
```
Comment (eric-wieser) [inline doc/HOWTO_RELEASE.rst.txt]: Not sure this is super useful, but I suppose it at least identifies the issues for a future PR

Comment (mwtoews) [reply doc/HOWTO_RELEASE.rst.txt]: Or, if anyone wants to suggest a better link, no better time than now.

**Comment by charris** [comment]
Thanks @mwtoews .

File: doc/HOWTO_RELEASE.rst.txt (hunk 1 above)
Comment (mattip) [reply doc/HOWTO_RELEASE.rst.txt]: This [email exchange](http://scipy-user.10969.n7.nabble.com/Building-extensions-on-Win64-with-GNU-compilers-td3156.html) suggests it should be removed

Comment (rgommers) [reply doc/HOWTO_RELEASE.rst.txt]: All the dead links as well as references to SourceForge can be removed.

**Pull request merged**